File: src/main/java/com/example/todotechproject/modelo/dto/ProductoDTO.java
package com.example.todotechproject.modelo.dto;

import jakarta.validation.constraints.NotNull;

public record ProductoDTO(Long id,
                          @NotNull(message = "Nombre is required")
                          String nombre,
                          String codigo,
                          String descripcion,
                          CategoriaDTO categoria,
                          Double precio,
                          Integer stock) {

}

File: frontend/src/app/models/producto.dto.ts
export interface CategoriaDTO {
    id: number;
    nombre: string;
    descripcion: string;
}

export interface ProductoDTO {
    id: number;
    nombre: string;
    codigo: string;
    descripcion: string;
    categoria: CategoriaDTO;
    precio: number;
    stock: number;
}